great_expectations/rule_based_profiler/domain_builder/simple_column_suffix_domain_builder.py

```diff
@@ -43,12 +57,15 @@ def _get_domains(
                     "Unrecognized column_name_suffixes directive -- must be a list or a string."
                 )
 
-        table_column_names: List[str] = validator.get_metric(
+        metric_domain_kwargs: dict = {
+            "batch_id": self.get_batch_id(variables=variables),
+        }
```

Comment:
This is maybe a philosophical question, and maybe out of scope: Can we always assume that a Batch is tabular? Currently, a `batch_id` is sufficient as the only key passed into `metric_domain_kwargs` because we know that a `batch_id` represents a clear single table. But if in a future world a Batch might be able to take other shapes, span multiple tables, etc. I wonder if we need some sort of method to validate that a Batch is a table so that we know we can rely on this here.

Reply:
@talagluck This is a good question -- the notion of what constitutes a batch has been going through constant stream of questions.  Because they have not been answered definitively yet, I believe that we can leave the implementation as is for now.  However, here are some of the questions that should be addressed in the coming months/quarters:
* Are we only considering tabular data?  If not, what other kind of data should we figure out how to represent?
* When we partition a large physical table, do we end up with one batch consisting of multiple partitions?  Or is each partition actually a batch?  (This is a very important question, because it focuses the discussion on what is being validated.  In other words, the user wants to validate a table, which happens to be large; hence, the `Validator` should represent that large table; however, if a batch represents a smaller partition of that table and the `Validator` represents that batch, then how do we help the user validate the actual large table of import to the user?)
* Why did we remove `batch_id` from the `TableDomainBuilder`?  If the user can now specify the `batch_request` in the domain builder's configuration, then why would we take away the ability of the users to build the domain out of any batch that interests them?  Since the `Validator` will accept an explicit `batch_id`, then the `TableDomainBuilder` should support the ability to set that `batch_id` from the `Batch` returned by processing the configured batch_request.  (I propose that we add this back in a quick follow on PR.)

Reply:
These are great questions!

To the last one, please check my understanding: `TableDomainBuilder` feels like a pretty specific case. As a reminder, `DomainBuilders` determine the DomainKwargs to use for building an ExpectationConfiguration. But table-level expectations don't need to take DomainKwargs - since Validators can currently only validate one Batch at a time, we just assume that a table-level expectation applies to the entire Batch. So `TableDomainBuilder` winds up being almost like a dummy Domain builder, saying the Domain for this ExpectationConfiguration is the entire Batch. From my understanding, creating an ExpectationConfiguration where a `batch_id` is passed in as a `DomainKwarg` will make it so that that expectation will only be validated when run on a Batch with that `batch_id` (which seems like an unusual case that most people wouldn't want). So I think to avoid that, we should keep this as is, and if we think that there is a use case for building a TableDomain for a specific `batch_id`, we should add a new DomainBuilder.